### DISCUSSION TITLE
Proposal to kickstart the creation and maintenance of a community maintained helm chart.

### DISCUSSION THREAD

**Discussion by conall88** [hoppscotch/hoppscotch#3402] [RFC]
Hi,

I'm currently devoting some time each week to creating a helm chart for Hoppscotch.io community edition. 

I was initially planning to develop this gradually, and offer it as-is once the initial version is what I would consider "fit for production".
But this seems quite inefficient when there are likely people out that that would be happy to contribute to it's progress and success.

during my brief time on discord, I asked the general chat if there was any demand for this, and it appears the answer is yes.

right now, I'm:

- initially targeting the standalone container releases tagged "2023.8.1"
- targeting Kubernetes 1.19.0 or later (although I may revise this to non-End of Life releases as per: https://kubernetes.io/releases/ ) 
- including the [bitnami postgres charts ](https://bitnami.com/stack/postgresql/helm)as a dependent subchart , an optional inclusion for those that don't already have a pgsql backend they want to use, and want to stand-one up as part of their deployment.

I have a few questions:

-are there any plans to create and support this functionality already?
-If I was to contribute this to the project, would the hoppscotch project prefer this to be PR'd into the hoppscotch/hoppscotch repo , or somewhere else?
-Is anyone familiar of the licensing ramifications of subchart inclusion? (If not I'm happy to research this and reach out to bitnami to discuss.)
-If I did contribute this templating, I would like to remain as a chart maintainer , would this be acceptable? :)

About me:
-I'm a support engineer at  https://grafana.com/    [github: [Grafana ](https://github.com/grafana/grafana)  ]
-I'm a Kubernetes Certified Administrator [ [link to credentials](https://www.credly.com/badges/4cc530f2-a7e7-41b3-8158-3fb154ddc8f3)]
-I'm trying to advocate the use of hoppscotch.io  internally, and deploying to kubes is the best fit for our org.

**Comment by Andreas-Schoenefeldt** [comment]
It looks like there is https://artifacthub.io/packages/helm/deliveryhero/hoppscotch, but the vulnarabilties of the chart are too high for an actual production usage:

SECURITY REPORT
414 vulnerabilities found
38 CRITICAL
190 HIGH
171 MEDIUM
15 LOW

**Reply by Andreas-Schoenefeldt** [reply]
https://github.com/WDaan/hoppscotch-helm looks newer

**Reply by conall88** [reply]
right the deliveryhero chart predates the hoppscotch re-arch, and is lacking many common features that I'd expect from a helm chart , such as support for various sql backends as subcharts, HPA, HA, etc.

I'l see if I can discuss with WDaan, perhaps I can merge some of my efforts into his project, or vice versa. 

I'l probably publish my current state to a public repo this month and go from there.

**Reply by daniil-titov-indriver** [reply]
hi, @conall88 !
do you have any updates?
I'm facing the same problem as you: there is no existing production ready way to deploy hoppscotch community in k8s for internal use

**Comment by conall88** [comment]
Hi daniil, 

I've shelved my efforts for the moment.

To define something as production ready would involve having some maintainers, rather than a single point of failure. I don't really have bandwidth to maintain an effort on my own yet, and if the root project isn't interested in getting involved, it does not bode well for the longevity of the project. I'd rather do something right , or not at all.

I hope opinions will change on this in the future.

In the interim, I think the best option is to deploy via docker compose.

I will leave this conversation open in the hopes things change. 

If you are reading this and feel dissapointed about the absence of proper helm charts for hoppscotch, please leave your feedback below so we can better express the demand for this to the parent project.
